crates/lib/src/rules/structure/ST07.rs

```diff
@@ -0,0 +1,244 @@
+use ahash::AHashMap;
+use itertools::Itertools;
+use smol_str::{SmolStr, ToSmolStr};
+
+use crate::core::config::Value;
+use crate::core::parser::segments::base::{
+    CodeSegmentNewArgs, ErasedSegment, IdentifierSegment, SymbolSegment, SymbolSegmentNewArgs,
+    WhitespaceSegment, WhitespaceSegmentNewArgs,
+};
+use crate::core::parser::segments::keyword::KeywordSegment;
+use crate::core::rules::base::{CloneRule, ErasedRule, LintFix, LintResult, Rule, RuleGroups};
+use crate::core::rules::context::RuleContext;
+use crate::core::rules::crawlers::{Crawler, SegmentSeekerCrawler};
+use crate::dialects::ansi::Node;
+use crate::dialects::SyntaxKind;
+use crate::helpers::ToErasedSegment;
+use crate::utils::analysis::select::get_select_statement_info;
+use crate::utils::functional::context::FunctionalContext;
+use crate::utils::functional::segments::Segments;
+
+#[derive(Clone, Debug, Default)]
+pub struct RuleST07;
+
+impl Rule for RuleST07 {
+    fn load_from_config(&self, _config: &AHashMap<String, Value>) -> Result<ErasedRule, String> {
+        Ok(RuleST07.erased())
+    }
+
+    fn is_fix_compatible(&self) -> bool {
+        true
+    }
+
+    fn name(&self) -> &'static str {
+        "structure.using"
+    }
+
+    fn description(&self) -> &'static str {
+        "Prefer specifying join keys instead of using ``USING``."
+    }
+
+    fn long_description(&self) -> &'static str {
+        r"
+**Anti-pattern**
+
+```sql
+SELECT
+    table_a.field_1,
+    table_b.field_2
+FROM
+    table_a
+INNER JOIN table_b USING (id)
+```
+
+**Best practice**
+
+Specify the keys directly
+
+```sql
+SELECT
+    table_a.field_1,
+    table_b.field_2
+FROM
+    table_a
+INNER JOIN table_b
+    ON table_a.id = table_b.id
+```"
+    }
+
+    fn groups(&self) -> &'static [RuleGroups] {
+        &[RuleGroups::All, RuleGroups::Structure]
+    }
+
+    fn eval(&self, context: RuleContext) -> Vec<LintResult> {
+        if context.dialect.name == "clickhouse" {
```

Comment:
nit: This feels a bit off here. It's worth maybe adding a FIXME/TODO, as it is something we may want to do more uniformly in the future. 

```suggestion
        // TODO Figure out how to make this cleaner and part of the definition rules
        if context.dialect.name == "clickhouse" {
```